### DISCUSSION TITLE
PHAL request - CEC plugin

### DISCUSSION THREAD

**Discussion by JarbasAl** [OpenVoiceOS/ovos-buildroot#158] [Dev]
using https://github.com/trainman419/python-cec

make any screen power on if wake word is detected, standby after X configurable seconds

independently of the above a companion IOT plugin once the framework is ready should also be made, for setups where you want to expose the connected screen as a IOT device

**Comment by builderjer** [comment]
CEC is HDMI dependent, correct? So the device would have to be directly connected to a HDMI screen?

**Reply by JarbasAl** [reply]
that is my understanding yes, but not all HDMI will work, i think sometimes you may need a dongle to get the CEC implementation, haven't read much in depth about it, @AIIX should have some experience from bigscreen and can probably give more useful feedback

**Reply by AIIX** [reply]
Yes, CEC is HDMI dependent, but HDMI-CEC itself is manufacturer dependent, having a HDMI port is not enough, the main board has to support CEC. Intel / Nvidia / AMD do not support HDMI CEC on laptops / desktops for example even if there is a HDMI port on the graphics card, for those systems that do not support it you need to purchase a separate USB adaptor from Pulse Eight https://www.pulse-eight.com/p/104/usb-hdmi-cec-adapter. Raspberry PI does support HDMI-CEC on board, so adapter is not needed.

This is also applicable the other way around, the monitor / screen also depends on manufacturer implementing CEC, for example Waveshare / RPI display screens do not support CEC, CEC is generally more commonly found on TVs and Surround sound systems that connect to it.

**Reply by builderjer** [reply]
And each manufacturer usually calls it something different also
